benchmark/src/main/java/ai/timefold/solver/benchmark/impl/aggregator/BenchmarkAggregator.java

```diff
@@ -40,11 +48,145 @@ public void setBenchmarkReportConfig(BenchmarkReportConfig benchmarkReportConfig
+        return this.aggregate(allSingleBenchmarkResults, solverBenchmarkResultNameMap);
+    }
+
+    public File aggregateSelectedBenchmarksInUi(List<SingleBenchmarkResult> selectedSingleBenchmarkResults,
+            Map<SolverBenchmarkResult, String> solverBenchmarkResultNameMap) {
+        return this.aggregate(selectedSingleBenchmarkResults, solverBenchmarkResultNameMap);
+    }
+
+    public File aggregateSelectedBenchmarks(PlannerBenchmarkConfig benchmarkConfig, List<String> selectedDirectoryNames) {
+        File benchmarkDirectory = benchmarkConfig.getBenchmarkDirectory();
+        if (!benchmarkDirectory.exists() || !benchmarkDirectory.isDirectory()) {
+            throw new IllegalArgumentException("Benchmark directory does not exist: " + benchmarkDirectory);
+        }
+
+        // Read all benchmark results first, then filter
+        BenchmarkResultIO benchmarkResultIO = new BenchmarkResultIO();
+        List<PlannerBenchmarkResult> allPlannerResults = benchmarkResultIO.readPlannerBenchmarkResultList(benchmarkDirectory);
+
+        List<SingleBenchmarkResult> selectedSingleBenchmarkResults = new ArrayList<>();
+        Map<SolverBenchmarkResult, String> solverBenchmarkResultNameMap = new HashMap<>();
+
+        // Filter results based on selected directory names
+        for (PlannerBenchmarkResult plannerResult : allPlannerResults) {
+            String benchmarkDirectoryName = plannerResult.getBenchmarkReportDirectory().getName();
+
+            if (selectedDirectoryNames.contains(benchmarkDirectoryName)) {
+                LOGGER.info("Including benchmark results from directory: {}", benchmarkDirectoryName);
+
+                for (SolverBenchmarkResult solverResult : plannerResult.getSolverBenchmarkResultList()) {
+                    selectedSingleBenchmarkResults.addAll(solverResult.getSingleBenchmarkResultList());
+                    solverBenchmarkResultNameMap.put(solverResult, solverResult.getName());
+                }
+            }
+        }
+
+        if (selectedSingleBenchmarkResults.isEmpty()) {
+            throw new IllegalArgumentException(
+                    "No valid benchmark results found for selected directories: " + selectedDirectoryNames);
+        }
+
+        // Configure the aggregator instance
+        this.setBenchmarkDirectory(benchmarkDirectory);
+        this.setBenchmarkReportConfig(benchmarkConfig.getBenchmarkReportConfig());
+
+        // Perform the aggregation with selected results - returns HTML report file
+        return this.aggregate(selectedSingleBenchmarkResults, solverBenchmarkResultNameMap);
+    }
+
+    public List<String> getAvailableBenchmarkDirectories(PlannerBenchmarkConfig benchmarkConfig) {
+        File benchmarkDirectory = benchmarkConfig.getBenchmarkDirectory();
+        List<String> directories = new ArrayList<>();
+
+        if (!benchmarkDirectory.exists() || !benchmarkDirectory.isDirectory()) {
+            return directories;
```

Comment:
You can return `Collections.emptyList()` here.
Which allows you to only allocate `directories` later - moving line 144 to after this block.